### PR TITLE
Retriever Fixes

### PR DESCRIPTION
**Pull request by misha-chertushkin**
## Description
This PR adds the ability to disable relevancy score to increase latency. Also it modifies the retriever behaviour to achieve better results on the first run. 
This PR succeeds over the PR https://github.com/GoogleCloudPlatform/platform-gen-ai/pull/14 as it contains better implementation.

### CODE REVIEW

File: gen_ai/common/document_retriever.py [hunk 1]
```diff
@@ -111,10 +111,12 @@ def get_related_docs_from_store(
         if metadata is not None and len(metadata) > 1:
             metadata = convert_to_chroma_format(metadata)
 
-        ss_docs = store.similarity_search(query=questions_for_search, k=5, filter=metadata)
+        ss_docs = store.similarity_search_with_score(query=questions_for_search, k=50, filter=metadata)
+        ss_docs = ss_docs[0:3]
+        ss_docs = [x[0] for x in ss_docs]
```
Comment (daureng) [inline gen_ai/common/document_retriever.py:116]: ss_docs = [x[0] for x in ss_docs[:3]]

**Review by daureng — APPROVED** [review]
lgtm, please check the comment

File: gen_ai/common/document_retriever.py (hunk 1 above)
Comment (anna-novakovska) [reply gen_ai/common/document_retriever.py:116]: @misha-chertushkin did you decide not to implement this Dauren's suggestion?

Comment (misha-chertushkin) [reply gen_ai/common/document_retriever.py]: I actually implemented it, see my last commit :)

**Review by anna-novakovska — APPROVED** [review]

**Pull request merged**